packages/hoppscotch-sh-admin/src/components/setup/DataSharingAndNewsletter.vue

```diff
@@ -0,0 +1,136 @@
+        </HoppSmartToggle>
+        <HoppSmartAnchor
+          blank
+          to="http://docs.hoppscotch.io"
+          :label="t('data_sharing.see_shared')"
+          class="underline"
+        />
+      </div>
+      <div class="flex flex-col items-start space-y-5">
+        <span>
+          <p class="text-lg font-bold text-white">
+            {{ t('newsletter.title') }}
+          </p>
+          <p class="font-light">{{ t('newsletter.description') }}</p>
+        </span>
+        <HoppSmartToggle
+          :on="newsletterToggle"
+          @change="newsletterToggle = !newsletterToggle"
+        >
+          {{ t('newsletter.toggle_description') }}
+        </HoppSmartToggle>
+      </div>
+      <div class="flex flex-col space-y-5">
+        <HoppButtonPrimary
+          :icon="IconLogIn"
+          :label="t('app.continue_to_dashboard')"
+          class="mx-10"
+          @click="submitSelection"
+        />
+        <HoppSmartAnchor
+          blank
+          to="http://docs.hoppscotch.io"
+          :icon="IconBookOpenText"
+          :label="t('app.read_documentation')"
+        />
+      </div>
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { useMutation } from '@urql/vue';
+import { ref } from 'vue';
+import { useI18n } from '~/composables/i18n';
+import { useToast } from '~/composables/toast';
+import { auth } from '~/helpers/auth';
+import { listmonkApi } from '~/helpers/axiosConfig';
+import {
+  ToggleAnalyticsCollectionDocument,
+  ToggleAnalyticsCollectionMutationVariables,
+} from '~/helpers/backend/graphql';
+import IconBookOpenText from '~icons/lucide/book-open-text';
+import IconLogIn from '~icons/lucide/log-in';
+
+const t = useI18n();
+const toast = useToast();
+const user = auth.getCurrentUser();
+
+const emit = defineEmits<{
+  (event: 'onSetupComplete', status: boolean): void;
+}>();
+
+const dataSharingToggle = ref(true);
+const newsletterToggle = ref(true);
+
+// Toggle data sharing
+const dataSharingMutation = useMutation(ToggleAnalyticsCollectionDocument);
+
+const toggleDataSharing = async () => {
+  const status = dataSharingToggle.value ? 'ENABLE' : 'DISABLE';
+
+  const variables = { status };
+  const result = await dataSharingMutation.executeMutation(
+    variables as ToggleAnalyticsCollectionMutationVariables
+  );
+
+  if (result.error) {
+    toast.error(t('state.data_sharing_failure'));
+    return false;
+  }
+  return true;
+};
+
+// Toggle subscription to newsletter
+const toggleNewsletter = async () => {
+  try {
+    await listmonkApi.post('/subscription', {
+      email: user?.email,
+      name: user?.displayName,
+      list_uuids: ['f5f0b457-44d0-4aa1-b6f9-165dc1efa56a'],
+    });
+
+    return true;
+  } catch (e) {
+    console.error(e);
+    toast.error(t('state.newsletter_failure'));
+    return false;
+  }
+};
+
+// Submit selections made
+const submitSelection = async () => {
+  const dataSharingResult = await toggleDataSharing();
```

Comment:
@jamesgeorge007 Based on discussion with the backend team, on first setup, data sharing will be disabled. So a mutation should ideally be fired only when it's toggled on. I implemented it in that way before. But I had a thought to fire the mutation with the new value of the toggle regardless of its initial value thereby not assuming whether data sharing is enabled or disabled by default. This is risk free. What do you recommend in this situation?

Reply:
Since we're assured that the default state is disabled we could trigger the GQL mutation only if the user opts in similar to the approach with the newsletter subscription. The network call in such a scenario is unnecessary IMO.